src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year2]) {
+    dateData[year2] = dateData[year4].slice(2, 4);
+  }
+
+  if (!dateData[year4]) {
+    if (dateData[year2] <= 20) {
+      dateData[year4] = `20${dateData[year2]}`;
+    } else {
+      dateData[year4] = `19${dateData[year2]}`;
+    }
+  }
```

Comment:
The condition to check for `dateData[year4]` and the subsequent logic to derive `year4` from `year2` should be inside the loop where you're building the result. This ensures that the conversion only happens if `YYYY` is part of the `toFormat`. Additionally, the comparison `dateData[year2] <= 20` should use a strict comparison `dateData[year2] < '30'` to include all years less than '30', not '20'. Remember to compare strings, since the values are not converted to numbers.